File: t-tasks/src/main/java/com/teo/ttasks/util/AnimUtils.java
package com.teo.ttasks.util;

import android.annotation.TargetApi;
import android.os.Build;
import android.support.v4.view.ViewCompat;
import android.support.v4.view.ViewPropertyAnimatorCompat;
import android.support.v4.view.animation.FastOutSlowInInterpolator;
import android.transition.Transition;
import android.view.View;
import android.view.animation.Interpolator;

public class AnimUtils {

    private AnimUtils() { }

    public static class TaskDetailAnim {

        private static final Interpolator interpolator = new FastOutSlowInInterpolator();
        private static final int duration = 200;
        private static final int delay = 200;

        public static ViewPropertyAnimatorCompat animate(View view) {
            return ViewCompat.animate(view)
                    .alpha(1f)
                    .setDuration(duration)
                    .setStartDelay(delay)
                    .setInterpolator(interpolator);
        }
    }

    @TargetApi(Build.VERSION_CODES.KITKAT)
    public static class TransitionListener implements Transition.TransitionListener {
        @Override public void onTransitionStart(Transition transition) {

        }

        @Override public void onTransitionEnd(Transition transition) {

        }

        @Override public void onTransitionCancel(Transition transition) {

        }

        @Override public void onTransitionPause(Transition transition) {

        }

        @Override public void onTransitionResume(Transition transition) {

        }
    }
}

File: t-tasks/src/main/java/com/teo/ttasks/util/AnimUtils.kt
package com.teo.ttasks.util

import android.annotation.TargetApi
import android.os.Build
import androidx.core.view.ViewCompat
import androidx.core.view.ViewPropertyAnimatorCompat
import android.transition.Transition
import android.view.View
import androidx.interpolator.view.animation.FastOutSlowInInterpolator

object AnimUtils {

    object TaskDetailAnim {

        private val interpolator = FastOutSlowInInterpolator()
        private const val duration = 200
        private const val delay = 200

        fun animate(view: View): ViewPropertyAnimatorCompat {
            return ViewCompat.animate(view)
                    .alpha(1f)
                    .setDuration(duration.toLong())
                    .setStartDelay(delay.toLong())
                    .setInterpolator(interpolator)
        }
    }

    @TargetApi(Build.VERSION_CODES.KITKAT)
    open class TransitionListener : Transition.TransitionListener {
        override fun onTransitionStart(transition: Transition) {

        }

        override fun onTransitionEnd(transition: Transition) {

        }

        override fun onTransitionCancel(transition: Transition) {

        }

        override fun onTransitionPause(transition: Transition) {

        }

        override fun onTransitionResume(transition: Transition) {

        }
    }
}
